### ISSUE TITLE
When influxdb config param set to "ping-auth-enabled = true", chronograf can not access /health endpoint (self-signed certs)

### ISSUE THREAD

**Issue by earthquakesan** [influxdata/chronograf#5545]
# Component versions

```
# Software Versions
kapacitor_version: "1.5"
influxdb_version: "1.8"
telegraf_version: "1.14"
chronograf_version: "1.7.17"
```

Everything is running as docker images. Deployed using ansible.

# Configurations

## Defaults

```
# Data Directories
data_home: "/data"
kapacitor_home: "{{ data_home }}/kapacitor"
influxdb_home: "{{ data_home }}/influxdb"
telegraf_home: "{{ data_home }}/telegraf"
chronograf_home: "{{ data_home }}/chronograf"

# Docker Network for Master Node
master_docker_network_name: "influxdb"

# Internal Docker Data Directories
influxdb_docker_data_dir: "/var/lib/influxdb"
chronograf_docker_data_dir: "/var/lib/chronograf"
kapacitor_docker_data_dir: "/var/lib/kapacitor"

# Internal Docker Configuration Files
# influxdb can be also configured using environment variables: https://docs.influxdata.com/influxdb/v1.8/administration/config/
# For AuthN/AuthZ: https://docs.influxdata.com/influxdb/v1.8/administration/authentication_and_authorization/
influxdb_docker_conf_file: "/etc/influxdb/influxdb.conf"
# chronograf is configured via env variables: https://docs.influxdata.com/chronograf/v1.8/administration/config-options/
# chronograf should be configured to login via gitlab https://docs.influxdata.com/chronograf/v1.8/administration/managing-security/#configure-generic-authentication
kapacitor_docker_conf_file: "/etc/kapacitor/kapacitor.conf"
telegraf_docker_conf_file: "/etc/telegraf/telegraf.conf"

# Software Versions
kapacitor_version: "1.5"
influxdb_version: "1.8"
telegraf_version: "1.14"
chronograf_version: "1.7.17"

# InfluxDB Configuration
influxdb_docker_port: "8088"
influxdb_port: "8088"
influxdb_http_docker_port: "8086"
influxdb_http_port: "8086"
influxdb_container_name: "influxdb"
influxdb_shared_secret: "snip"
influxdb_admin_username: "admin"
influxdb_admin_password: "admin"

# Kapacitor Configuration
kapacitor_container_name: "kapacitor"
kapacitor_hostname: "snip"
kapacitor_docker_port: "9092"
kapacitor_port: "9092"
## pagerduty is the main alerts channel
kapacitor_pagerduty_api_key: "snip"

# Chronograf Configuration
chronograf_container_name: "chronograf"
chronograf_docker_port: "8888"
chronograf_port: "8888"
chronograf_organization: "default"
chronograf_canned_path: "/usr/share/chronograf/canned"
# status_feed should be from the same domain as website, otherwise will get blocked
chronograf_status_feed: "snip"

# Telegraf Configuration
telegraf_container_name: "telegraf"
influxdb_telegraf_database: "telegraf"
# use hostname as defined in inventory
telegraf_hostname: "{{ inventory_hostname }}"
influxdb_public_hostname: "10.10.10.10"
telegraf_data_collection_interval: "10s"
telegraf_username: "telegraf"
telegraf_password: "telegraf"
```

## Chronograf config

Set via enviroment variables of docker container

```
    env:
      HOST: "0.0.0.0"
      PORT: "{{ chronograf_docker_port }}" # 8888
      BOLT_PATH: "{{ chronograf_docker_data_dir }}/database/chronograf-v1.db"
      STATUS_FEED_URL: "{{ chronograf_status_feed }}"
      GENERIC_NAME: "gitlab"
      GENERIC_CLIENT_ID: "snip"
      GENERIC_CLIENT_SECRET: "snip"
      GENERIC_AUTH_URL: "https://snip/oauth/authorize"
      GENERIC_TOKEN_URL: "https://snip/oauth/token"
      TOKEN_SECRET: "snip"
      GENERIC_SCOPES: "openid,read_user"
      PUBLIC_URL: "https://snip"
      GENERIC_API_URL: "https://snip/api/v3/user"
      REPORTING_DISABLED: "true"
      RESOURCES_PATH: "{{ chronograf_docker_data_dir }}/resources"
```

## InfluxDB config

Rendered as file and mounted into the docker container.

```
reporting-disabled = true
bind-address = "127.0.0.1:{{ influxdb_docker_port }}"

[meta]
  dir = "/var/lib/influxdb/meta"
  retention-autocreate = true
  logging-enabled = true

[data]
  dir = "/var/lib/influxdb/data"
  index-version = "inmem"
  wal-dir = "/var/lib/influxdb/wal"
  wal-fsync-delay = "0s"
  validate-keys = false
  query-log-enabled = true
  cache-max-memory-size = 1073741824
  cache-snapshot-memory-size = 26214400
  cache-snapshot-write-cold-duration = "10m0s"
  compact-full-write-cold-duration = "4h0m0s"
  compact-throughput = 50331648
  compact-throughput-burst = 50331648
  max-series-per-database = 1000000
  max-values-per-tag = 100000
  max-concurrent-compactions = 0
  max-index-log-file-size = 1048576
  series-id-set-cache-size = 100
  series-file-max-concurrent-snapshot-compactions = 0
  trace-logging-enabled = false
  tsm-use-madv-willneed = false

[coordinator]
  write-timeout = "10s"
  max-concurrent-queries = 0
  query-timeout = "0s"
  log-queries-after = "0s"
  max-select-point = 0
  max-select-series = 0
  max-select-buckets = 0

[retention]
  enabled = true
  check-interval = "30m0s"

[shard-precreation]
  enabled = true
  check-interval = "10m0s"
  advance-period = "30m0s"

[monitor]
  store-enabled = true
  store-database = "_internal"
  store-interval = "10s"

[subscriber]
  enabled = true
  http-timeout = "30s"
  insecure-skip-verify = false
  ca-certs = ""
  write-concurrency = 40
  write-buffer-size = 1000

[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = true
  log-enabled = true
  suppress-write-log = false
  write-tracing = false
  flux-enabled = false
  flux-log-enabled = false
  pprof-enabled = true
  pprof-auth-enabled = false
  debug-pprof-enabled = false
  ping-auth-enabled = true
  https-enabled = true
  https-certificate = "{{ influxdb_docker_data_dir }}/server-cert.pem"
  https-private-key = "{{ influxdb_docker_data_dir }}/server-key.pem"
  max-row-limit = 0
  max-connection-limit = 0
  shared-secret = "{{ influxdb_shared_secret }}"
  realm = "InfluxDB"
  unix-socket-enabled = false
  unix-socket-permissions = "0777"
  bind-socket = "/var/run/influxdb.sock"
  max-body-size = 25000000
  access-log-path = ""
  max-concurrent-write-limit = 0
  max-enqueued-write-limit = 0
  enqueued-write-timeout = 30000000000

[logging]
  format = "auto"
  level = "info"
  suppress-logo = false

[[graphite]]
  enabled = false
  bind-address = ":2003"
  database = "graphite"
  retention-policy = ""
  protocol = "tcp"
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "1s"
  consistency-level = "one"
  separator = "."
  udp-read-buffer = 0

[[collectd]]
  enabled = false
  bind-address = ":25826"
  database = "collectd"
  retention-policy = ""
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "10s"
  read-buffer = 0
  typesdb = "/usr/share/collectd/types.db"
  security-level = "none"
  auth-file = "/etc/collectd/auth_file"
  parse-multivalue-plugin = "split"

[[opentsdb]]
  enabled = false
  bind-address = ":4242"
  database = "opentsdb"
  retention-policy = ""
  consistency-level = "one"
  tls-enabled = false
  certificate = "/etc/ssl/influxdb.pem"
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "1s"
  log-point-errors = true

[[udp]]
  enabled = false
  bind-address = ":8089"
  database = "udp"
  retention-policy = ""
  batch-size = 5000
  batch-pending = 10
  read-buffer = 0
  batch-timeout = "1s"
  precision = ""

[continuous_queries]
  log-enabled = true
  enabled = true
  query-stats-enabled = false
  run-interval = "1s"

[tls]
  min-version = ""
  max-version = ""
```

## Kapacitor Config

Rendered as file and mounted into the docker container.

```
hostname = "{{ kapacitor_hostname }}"
data_dir = "/var/lib/kapacitor"
skip-config-overrides = false
default-retention-policy = ""

[alert]
  persist-topics = true

[http]
  bind-address = ":{{ kapacitor_docker_port }}"
  auth-enabled = false
  log-enabled = true
  write-tracing = false
  pprof-enabled = false
  https-enabled = false
  https-certificate = "/etc/ssl/kapacitor.pem"
  https-private-key = ""
  shutdown-timeout = "10s"
  shared-secret = ""

[replay]
  dir = "/var/lib/kapacitor/replay"

[storage]
  boltdb = "/var/lib/kapacitor/kapacitor.db"

[task]
  dir = "/root/.kapacitor/tasks"
  snapshot-interval = "1m0s"

[load]
  enabled = false
  dir = "/root/.kapacitor/load"

[[influxdb]]
  enabled = true
  name = "snip"
  default = true
  urls = ["https://{{ influxdb_container_name }}:{{ influxdb_http_docker_port }}"]
  username = "{{ influxdb_admin_username }}"
  password = "{{ influxdb_admin_password }}"
  ssl-ca = ""
  ssl-cert = ""
  ssl-key = ""
  insecure-skip-verify = true
  timeout = "0s"
  disable-subscriptions = false
  subscription-protocol = "https"
  subscription-mode = "cluster"
  kapacitor-hostname = ""
  http-port = 0
  udp-bind = ""
  udp-buffer = 1000
  udp-read-buffer = 0
  startup-timeout = "5m0s"
  subscriptions-sync-interval = "1m0s"
  [influxdb.excluded-subscriptions]
    _kapacitor = ["autogen"]

[logging]
  file = "STDERR"
  level = "INFO"

[config-override]
  enabled = true

[tls]
  min-version = ""
  max-version = ""

[collectd]
  enabled = false
  bind-address = ":25826"
  database = "collectd"
  retention-policy = ""
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "10s"
  read-buffer = 0
  typesdb = "/usr/share/collectd/types.db"

[opentsdb]
  enabled = false
  bind-address = ":4242"
  database = "opentsdb"
  retention-policy = ""
  consistency-level = "one"
  tls-enabled = false
  certificate = "/etc/ssl/influxdb.pem"
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "1s"
  log-point-errors = true

[alerta]
  enabled = false
  url = ""
  insecure-skip-verify = false
  token = ""
  token-prefix = ""
  environment = ""
  origin = ""
  timeout = "0s"

[hipchat]
  enabled = false
  url = ""
  token = ""
  room = ""
  global = false
  state-changes-only = false

[[kafka]]
  enabled = false
  id = "default"
  timeout = "0s"
  batch-size = 0
  batch-timeout = "0s"
  use-ssl = false
  ssl-ca = ""
  ssl-cert = ""
  ssl-key = ""
  insecure-skip-verify = false

[[mqtt]]
  enabled = false
  name = "default"
  default = false
  url = ""
  ssl-ca = ""
  ssl-cert = ""
  ssl-key = ""
  insecure-skip-verify = false
  client-id = ""
  username = ""
  password = ""

[opsgenie]
  enabled = false
  api-key = ""
  url = "https://api.opsgenie.com/v1/json/alert"
  recovery_url = "https://api.opsgenie.com/v1/json/alert/note"
  global = false

[opsgenie2]
  enabled = false
  api-key = ""
  url = "https://api.opsgenie.com/v2/alerts"
  recovery_action = "notes"
  global = false

[pagerduty]
  enabled = false
  url = "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
  service-key = ""
  global = false

[pagerduty2]
  enabled = true
  url = "https://events.pagerduty.com/v2/enqueue"
  routing-key = "{{ kapacitor_pagerduty_api_key }}"
  global = true

[pushover]
  enabled = false
  token = ""
  user-key = ""
  url = "https://api.pushover.net/1/messages.json"

[[httppost]]
  endpoint = "example"
  url = "http://example.com"
  alert-template = ""
  alert-template-file = ""
  row-template = ""
  row-template-file = ""
  [httppost.basic-auth]
    username = ""
    password = ""

[smtp]
  enabled = false
  host = "192.168.175.35"
  port = 587
  username = "{{ kapacitor_smtp_username }}"
  password = "{{ kapacitor_smtp_password }}"
  no-verify = false
  global = false
  state-changes-only = false
  from = "{{ kapacitor_smtp_from }}"
  idle-timeout = "30s"
  to = {{ kapacitor_smtp_to }}

[snmptrap]
  enabled = false
  addr = "localhost:162"
  community = "kapacitor"
  retries = 1

[sensu]
  enabled = false
  addr = ""
  source = "Kapacitor"

[[slack]]
  enabled = false
  default = true
  workspace = ""
  url = ""
  channel = ""
  username = "kapacitor"
  icon-emoji = ""
  global = false
  state-changes-only = false
  ssl-ca = ""
  ssl-cert = ""
  ssl-key = ""
  insecure-skip-verify = false

[talk]
  enabled = false
  url = ""
  author_name = ""

[telegram]
  enabled = false
  url = "https://api.telegram.org/bot"
  token = ""
  chat-id = ""
  parse-mode = ""
  disable-web-page-preview = false
  disable-notification = false
  global = false
  state-changes-only = false

[victorops]
  enabled = false
  api-key = ""
  routing-key = ""
  url = "https://alert.victorops.com/integrations/generic/20131114/alert"
  global = false
  json-data = false

[reporting]
  enabled = false
  url = "https://usage.influxdata.com"

[stats]
  enabled = true
  stats-interval = "10s"
  database = "_kapacitor"
  retention-policy = "autogen"
  timing-sample-rate = 0.1
  timing-movavg-size = 1000

[udf]
```

## Kapacitor kap file

Rendered as file and mounted into the docker container.

```
{
  "id": "10001",
  "srcID": "10000",
  "name": "snip Kapacitor",
  "url": "http://{{ kapacitor_container_name }}:{{ kapacitor_docker_port }}",
  "active": true,
  "organization": "{{ chronograf_organization }}"
}
```

## InfluxDB src file

Rendered as file and mounted into the docker container.

```
{
  "id": "10000",
  "name": "snip InfluxDB",
  "username": "{{ influxdb_admin_username }}",
  "password": "{{ influxdb_admin_password }}",
  "url": "https://{{ influxdb_container_name }}:{{ influxdb_http_docker_port }}",
  "type": "influx",
  "insecureSkipVerify": true,
  "default": true,
  "telegraf": "{{ influxdb_telegraf_database }}",
  "organization": "{{ chronograf_organization }}"
}
```

# Expected behaviour

When the `ping-auth-enabled = true` set, the request to health endpoint should be successful. In the browser console I can see a GET request to: `https://snip/chronograf/v1/sources/10000/health` (204)

# Current behaviour

When the `ping-auth-enabled = true` set, the request to health endpoint fails. In the browser console I can see a GET request to: `https://snip/chronograf/v1/sources/10000/health` (401).

# Additional information

The certificates are self-signed.

# Current solution

Don't set this flag to true (not secure).

**Comment by russorat** [comment]
@earthquakesan thanks for the issue. I believe this was fixed via: https://github.com/influxdata/chronograf/pull/5430

Can you upgrade chronograf to 1.8.5 and try again?

**Comment by earthquakesan** [comment]
@russorat there is no docker image on hub.docker.com for 1.8.5. How long does it take since release until the images are published? (release was done 20 days ago).

**Comment by earthquakesan** [comment]
@russorat also, I can not use 1.8.4, because it is not able to connect to our corporate gitlab:

```
time="2020-07-27T20:59:23Z" level=error msg="Failed to validate Oauth settings: missing Google oauth setting[s]: client id, client secret" basepath=invalid component=server
```

That was the reason, why I am using 1.7.17 (it works just fine).

**Comment by russorat** [comment]
@earthquakesan thanks! i'm trying to track down why our docker images stopped updating. stay tuned.

**Comment by sranka** [comment]
This particular issue is already fixed via #5430. I also tried the latest chronograf 1.8.8 docker, it works fine. 

The Google OAuth setting is documented in https://docs.influxdata.com/chronograf/v1.8/administration/managing-security/#configure-google-authentication. Your OAuth error indicates that Google Auth was not set up properly (env: GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET were missing) ... it might have been caused by #5458 that is also already fixed. If you will still encounter Google OAuth configuration issues, please report another issue. Thank you.
